### PR TITLE
feat: add new type column to Feed entity

### PR DESCRIPTION
**Pull request by omBratteng**
Creates a new generated column `type` on `Feed` entity which is `main` when feed.id equals feed.userId, otherwise it is `custom`.

AS-837

### CODE REVIEW

File: src/migration/1734071060646-FeedTypeColumn.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class FeedTypeColumn1734071060646 implements MigrationInterface {
+  name = 'FeedTypeColumn1734071060646'
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`ALTER TABLE "feed" ADD "type" text GENERATED ALWAYS AS (CASE WHEN "id" = "userId" THEN 'main' ELSE 'custom' END) STORED NOT NULL`);
+  }
```
Comment (capJavert) [inline src/migration/1734071060646-FeedTypeColumn.ts:8]: I think usually there is insert to `typeorm_metadata` as well, see examples for other generated column migrations

File: src/entity/Feed.ts [hunk 2]
```diff
@@ -35,6 +40,17 @@ export class Feed {
...
+  @Column({
+    type: 'text',
+    update: false,
+    insert: false,
+    nullable: false,
+    unique: false,
+    generatedType: 'STORED',
+    asExpression: `CASE WHEN "id" = "userId" THEN 'main' ELSE 'custom' END`,
+  })
+  type: FeedType;
```
Comment (capJavert) [inline src/entity/Feed.ts:52]: hmm I thought to put it to graphorm, but I guess this works as well 👍

File: src/migration/1734071060646-FeedTypeColumn.ts (hunk 1 above)
Comment (omBratteng) [reply src/migration/1734071060646-FeedTypeColumn.ts]: Probably missed it, the auto generated migration is so noisy 🙈

File: src/entity/Feed.ts (hunk 2 above)
Comment (omBratteng) [reply src/entity/Feed.ts:52]: Didn't think of using GraphORM, though think it's nicer to keep it here, so could use it for for SQL stuff.

**Review by rebelchris — APPROVED** [review]

**Review by capJavert — APPROVED** [review]

**Pull request merged**